### ISSUE TITLE
RCS1169: "Mark field as read-only" analyzer triggers on modified public/protected/internal fields used in inheritors.

### ISSUE THREAD

**Issue by NickCraver** [JosefPihrt/Roslynator#64]
Best explained in a test case:
```c#
public class Base
{
    protected long _foo;
}
public class Child : Base
{
    public void Test() => Interlocked.Increment(ref _foo);
}
```
(simpler: even `_foo++` in a child class class does this)

I ran into this on a large refactor, the usage seems to be properly checked in the class itself, but not for uses in an inherited class. I tested `public`, `private`, and `internal` just to check - all have the same result here.

**Comment by NickCraver** [comment]
Adding another case I hit here, usage from an unrelated class on a static field, minimal repro:
```C#
public class Counters
{
    internal static long _foo;
}
public class Unrelated
{
    public void Test() => Interlocked.Increment(ref Counters._foo);
}
```

**Comment by JosefPihrt** [comment]
Fixed with #56
